tests/Integ/S3Context.php

```diff
@@ -280,4 +145,120 @@ public function iChangeTheBodyOfThePreSignedRequestToBe($body)
         $this->presignedRequest = $this->presignedRequest
             ->withBody(Psr7\stream_for($body));
     }
+
+
+    /**
+     * @Given I have an s3 client and I have a file
+     */
+    public function iHaveAnClientAndIHaveAFile()
+    {
+        $this->s3Client = self::getSdk()->createS3();
+        $this->stream = Psr7\stream_for(Psr7\try_fopen(self::$tempFile, 'r'));
+    }
+
+    /**
+     * @When I create a POST object with policy condition :version:
+     */
+    public function iCreateAPostObjectWithPolicyCondition(
+        $version,
+        TableNode $table
+    ){
+        date_default_timezone_set('UTC');
+
+        $policy = [
+            'expiration' => gmdate('Y-m-d\TH:i:s\Z', strtotime('+1 hours')),
+            'conditions' => [
+                ["bucket" => self::getResourceName()],
+                ["starts-with", '$key', ""],
+            ],
+        ];
+
+        $args = [];
+        foreach ($table as $row) {
+            $args[$row['key']] = $row['value'];
+        }
+
+        if ($version == 'v2') {
+            $policy['conditions'][] = json_decode($args['condition'], true);
+            $postObject = new PostObject(
+                $this->s3Client,
+                self::getResourceName(),
+                [
+                    'acl' => 'public-read'
+                ],
+                $policy
+            );
+
+            $this->preparePostData($postObject, 'v2');
+
+        } else { // v4
+            $ldt = gmdate(PostObjectV4::ISO8601_BASIC);
+            $policy['conditions'][] = ["X-Amz-Date" => $ldt];
+
+            $sdt = substr($ldt, 0, 8);
+            $scope = "$sdt/{$this->s3Client->getRegion()}/s3/aws4_request";
+            $credentials = "{$this->s3Client->getCredentials()->wait()->getAccessKeyId()}/$scope";
+            $policy['conditions'][] = ["X-Amz-Credential" => $credentials];
```

Comment:
The 'X-Amz-Credential' condition should be handled by the signer. As you mentioned when we discussed this in person, the condition field names are case sensitive, and that complexity should be encapsulated in the signer.